### PR TITLE
Move to modern setuptools with pyproject.toml

### PR DESCRIPTION
**Pull request by filmor**
Also moves the version to a common ̀`version.txt` that is read by both
the .NET and Python builds.

### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Ensure you have signed the [.NET Foundation CLA](https://cla.dotnetfoundation.org/pythonnet/pythonnet)
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by lostmsu** [comment]
> Python 3.6
>  No matching distribution found for setuptools>=61

How about we drop 3.6 [as suggested](https://github.com/pythonnet/pythonnet/pull/1653)? 8-)

File: Directory.Build.props [hunk 1]
```diff
@@ -1,11 +1,12 @@
+<?xml version="1.0"?>
 <Project>
   <PropertyGroup>
-    <VersionPrefix>3.0.0</VersionPrefix>
```
Comment (lostmsu) [inline Directory.Build.props:3]: `VersionPrefix` is necessary to create preview packages for NuGet

**Review by lostmsu — CHANGES_REQUESTED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (filmor) [reply Directory.Build.props:3]: I'd suggest that we just write the generated preview version to the `version.txt` with a small script.

File: Directory.Build.props [hunk 2]
```diff
@@ -6,7 +6,9 @@
     <AssemblyProduct>Python.NET</AssemblyProduct>
     <LangVersion>10.0</LangVersion>
     <IsPackable>false</IsPackable>
-    <Version>$([System.IO.File]::ReadAllText("version.txt"))</Version>
+    <FullVersion>$([System.IO.File]::ReadAllText("version.txt"))</FullVersion>
+    <VersionPrefix>$(FullVersion.Split('-', 2)[0])</VersionPrefix>
```
Comment (lostmsu) [inline Directory.Build.props:10]: Can you fix the monthly NuGet script right away? Should be easy to try locally

Comment (lostmsu) [reply Directory.Build.props:10]: Currently the version looks like this: `3.0.0-preview2022-04-11` 3.0.0 being the prefix and the rest is the suffix.

Comment (filmor) [reply Directory.Build.props:10]: I'm not sure I understand, with this change I get exactly the result you would expect. When specifying the suffix on the command-line, it overrides whatever suffix (if one was set) from the `version.txt` file.

**Review by lostmsu — APPROVED** [review]

**Pull request merged**